File: 0kth-largest-element-in-a-stream/0kth-largest-element-in-a-stream.cs
public class KthLargest {
    
    PriorityQueue<int,int> minheap=new PriorityQueue<int,int>();
    int Kval;
    
    public KthLargest(int k, int[] nums) {
        Kval=k;
        
        for(int i=0;i<nums.Length;i++)
        {
            minheap.Enqueue(nums[i],nums[i]);
            if(i>=Kval)
            {
                minheap.Dequeue();
            }
                
        }
    }
    
    public int Add(int val) {
        
        int kthlargestVal=0;
        minheap.Enqueue(val,val);
        if(minheap.Count>Kval)
        {
            minheap.Dequeue();
        }
        
        kthlargestVal=minheap.Peek();
        
        return kthlargestVal;
        
    }
}

/**
 * Your KthLargest object will be instantiated and called as such:
 * KthLargest obj = new KthLargest(k, nums);
 * int param_1 = obj.Add(val);
 */

File: 0kth-largest-element-in-a-stream/0kth-largest-element-in-a-stream.java
class KthLargest {
    
    PriorityQueue<Integer> queue= new PriorityQueue<Integer>();
    int kval;
    
    public KthLargest(int k, int[] nums) {
        kval=k;
        int i=0;
        
        if(nums.length==0)
        {
            return;
        }
        
        while(i<nums.length)
        {
            while(queue.size()>kval)
            {
                queue.remove();    
            }
            
            queue.add(nums[i]);
            i++;
        }
    }
    
    public int add(int val) {
        
        queue.add(val);
        
        while(queue.size()>kval)
        {
            queue.remove();
        }
        
        return queue.peek();
    }
}

/**
 * Your KthLargest object will be instantiated and called as such:
 * KthLargest obj = new KthLargest(k, nums);
 * int param_1 = obj.add(val);
 */